.github/workflows/plugin_test.yaml

```diff
@@ -0,0 +1,63 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: test plugin
+
+on:
+  push:
+    branches: [generic_test_in_ci_cd]
+  pull_request:
+    branches: [master]
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python_version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python_version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install coverage coveralls
+      - name: Install package
+        run: |
+          pip install ".[dev]"
+      - name: Install STM Plugin
+        run: |
+          git clone https://github.com/FAIRmat-NFDI/pynxtools-stm
+          cd pynxtools-stm
+          git checkout TestFromPynxtools
+          pip install .
+          pytest tests/test_reader.py::test_stm_reader
```

Comment:
I would split this into multiple steps in the action. Also I would use the [github checkout action template](https://github.com/actions/checkout). See the section [here](https://github.com/actions/checkout?tab=readme-ov-file#checkout-multiple-repos-side-by-side) on how to checkout multiple repos.

Reply:
Thanks. I check it, it is more robust. I will implement it. would you just update what types of function you want to test in pynxtools. Then, with the new approve I can check the ci/cd and prepare the PR for final review.

Reply:
Normally it should be fine if you just do `pytest .` on the plugins. That should gather all relevant tests and run them. That way the plugin writers have the greatest flexibility of implementing tests.